src/world/Units/Unit.cpp

```diff
@@ -76,167 +77,409 @@ void Unit::setClass(uint8_t class_) { write(unitData()->field_bytes_0.s.unit_cla
 uint8_t Unit::getGender() const { return unitData()->field_bytes_0.s.gender; }
 void Unit::setGender(uint8_t gender) { write(unitData()->field_bytes_0.s.gender, gender); }
 
-uint8_t Unit::getPowerType() const { return unitData()->field_bytes_0.s.power_type; }
-void Unit::setPowerType(uint8_t powerType) { write(unitData()->field_bytes_0.s.power_type, powerType); }
+PowerType Unit::getPowerType() const { return static_cast<PowerType>(unitData()->field_bytes_0.s.power_type); }
+void Unit::setPowerType(uint8_t powerType)
+{
+    write(unitData()->field_bytes_0.s.power_type, powerType);
+
+#ifdef AE_TBC
```

Comment:
Use basic VERSION_STRING check, AE_XYZ will be removed soon.

Reply:
I've updated them with VERSION_STRING checks